lib/bamboo/mailer.ex

```diff
@@ -62,19 +62,16 @@ defmodule Bamboo.Mailer do
 
   defmacro __using__(opts) do
     quote bind_quoted: [opts: opts] do
-      config = Bamboo.Mailer.parse_opts(__MODULE__, opts)
-
-      @adapter config.adapter
-      @config config
-
       @spec deliver_now(Bamboo.Email.t) :: Bamboo.Email.t
       def deliver_now(email) do
-        Bamboo.Mailer.deliver_now(@adapter, email, @config)
+        conf = config
+        Bamboo.Mailer.deliver_now(conf.adapter, email, conf)
       end
 
       @spec deliver_later(Bamboo.Email.t) :: Bamboo.Email.t
       def deliver_later(email) do
-        Bamboo.Mailer.deliver_later(@adapter, email, @config)
+        conf = config
+        Bamboo.Mailer.deliver_later(conf.adapter, email, conf)
```

Comment:
Rather than creating an intermediate variable `conf`, you could use `config` in the first and third argument. I think that might work. It might be *slightly* slower, but that is probably ok. Alternatively I would assign to `config` and rename the function to `get_config`